### PR TITLE
Fixes #25514 - Better error handling when sending from spool

### PR DESCRIPTION
**Pull request by xprazak2**

### CODE REVIEW

**Comment by xprazak2** [comment]
Proxy part for https://github.com/theforeman/foreman_openscap/pull/364

File: lib/smart_proxy_openscap/foreman_forwarder.rb [hunk 1]
```diff
@@ -9,14 +11,13 @@ def post_arf_report(cname, policy_id, date, data)
...
-        res = JSON.parse(response.body)
-        raise StandardError, "Received response: #{response.code} #{response.msg}" unless res['result'] == 'OK'
-      rescue StandardError => e
-        logger.debug response.body if response
-        logger.debug e.backtrace.join("\n\t")
+        JSON.parse(response.body)
+      rescue Net::HTTPServerException => e
+        logger.debug "Received response: #{response.code} #{response.msg}"
+        logger.debug response.body
+        raise ReportUploadError, e.message if response.code.to_i == 422
```
Comment (lzap) [inline lib/smart_proxy_openscap/foreman_forwarder.rb]: It would probably be safer to raise this when response code is not 200.

**Comment by lzap** [comment]
Other than that it looks fine.

File: lib/smart_proxy_openscap/foreman_forwarder.rb (hunk 1 above)
Comment (xprazak2) [reply lib/smart_proxy_openscap/foreman_forwarder.rb]: The exception does get re-raised, see the next line. I just need to distinguish the 422 case and handle it separately in `openscap_api`.

**Comment by ares** [comment]
Thanks @xprazak2 and @lzap, megring!

**Pull request merged**